### PR TITLE
Refactors ClassDetailViewController

### PR DESCRIPTION
**Pull request by kevinchan159**
- Refactors` ClassDetailViewController` so that instead of having all the views inside a `scrollView`, we split it up into a collection view
- Adds `StretchyHeaderLayout` that allows for the stretchy header effect
- Change organization name to `Cornell AppDev`

### CODE REVIEW

File: Fitness/Controllers/HabitTrackingController+Extensions.swift [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //  Fitness
 //
 //  Created by Kevin Chan on 5/14/19.
-//  Copyright © 2019 Keivan Shahida. All rights reserved.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
```
Comment (Omarrasheed) [inline Fitness/Controllers/HabitTrackingController+Extensions.swift]: lmao y was this made under keivan's name anyway

File: Fitness/Controllers/HomeController.swift [hunk 2]
```diff
@@ -2,7 +2,7 @@
 //  HomeController.swift
 //  Fitness
 //
-//  Created by Keivan Shahida on 2/24/18.
+//  Created by Cornell AppDev on 2/24/18.
```
Comment (Omarrasheed) [inline Fitness/Controllers/HomeController.swift]: Was this necessary?

File: Fitness/Extensions/String+Shared.swift [hunk 3]
```diff
@@ -6,10 +6,22 @@
 //  Copyright © 2018 Uplift. All rights reserved.
 //
 
-import Foundation
+import UIKit
```
Comment (Omarrasheed) [inline Fitness/Extensions/String+Shared.swift]: j curious, what's the dif?

File: Fitness/Views/ClassDetail/ClassDetailDescriptionCell.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    func configure(for gymClassInstance: GymClassInstance) {
+        descriptionTextView.text = gymClassInstance.classDescription
+    }
+
+    // MARK: - Private helpers
+    private func setupViews() {
+        descriptionTextView.font = ._14MontserratLight
+        descriptionTextView.isEditable = false
+        descriptionTextView.textAlignment = .center
+        descriptionTextView.sizeToFit()
```
Comment (Omarrasheed) [inline Fitness/Views/ClassDetail/ClassDetailDescriptionCell.swift]: is this necessary if ur already setting constraints elsewhere?

File: Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        descriptionLabel.text = gymClassInstance.tags.map { $0.name }.joined(separator: " · ")
+    }
+
+    // MARK: - Private helpers
+    private func setupViews() {
+        functionLabel.text = "FUNCTION"
+        functionLabel.font = ._16MontserratMedium
+        functionLabel.textAlignment = .center
+        functionLabel.textColor = .fitnessBlack
+        functionLabel.sizeToFit()
```
Comment (Omarrasheed) [inline Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift]: same

File: Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        contentView.addSubview(functionLabel)
+
+        descriptionLabel.font = ._14MontserratLight
+        descriptionLabel.textAlignment = .center
+        descriptionLabel.textColor = .fitnessBlack
+        descriptionLabel.sizeToFit()
```
Comment (Omarrasheed) [inline Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift]: same

File: Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            make.height.equalTo(descriptionLabelHeight)
+        }
+
+        dividerView.snp.makeConstraints { make in
+            make.height.equalTo(dividerViewHeight)
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview()
+            make.top.equalTo(descriptionLabel.snp.bottom).offset(dividerSpacing)
+        }
+
```
Comment (Omarrasheed) [inline Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift]: delete newline

File: Fitness/Views/ClassDetail/ClassDetailHeaderView.swift [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        addSubview(imageFilterView)
+
+        semicircleImageView.contentMode = UIView.ContentMode.scaleAspectFit
+        semicircleImageView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(semicircleImageView)
+
+        titleLabel.font = ._48Bebas
+        titleLabel.textAlignment = .center
+        titleLabel.textColor = .white
+        titleLabel.sizeToFit()
```
Comment (Omarrasheed) [inline Fitness/Views/ClassDetail/ClassDetailHeaderView.swift]: I feel like we can j take out all of the sizeToFit() calls but I also could be wrong

File: Podfile [hunk 9]
```diff
@@ -5,13 +5,14 @@ target 'Fitness' do
   # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
   use_frameworks!
   pod 'AlamofireImage'
+  pod 'Apollo', :git => 'https://github.com/apollographql/apollo-ios.git', :commit => 'b28c3dc'
+  pod 'Bartinter'
   pod 'Crashlytics'
+  pod 'FLEX', '~> 2.0', :configurations => ['Debug']
```
Comment (Omarrasheed) [inline Podfile]: Do we wanna push this to master/should we learn to use this on all pods?

File: Fitness/Controllers/HabitTrackingController+Extensions.swift (hunk 1 above)
Comment (kevinchan159) [reply Fitness/Controllers/HabitTrackingController+Extensions.swift]: i have no clue

File: Fitness/Controllers/HomeController.swift (hunk 2 above)
Comment (kevinchan159) [reply Fitness/Controllers/HomeController.swift]: I did a `Find and replace in workspace`

File: Fitness/Extensions/String+Shared.swift (hunk 3 above)
Comment (kevinchan159) [reply Fitness/Extensions/String+Shared.swift]: UIKit is what gives you all the UI components and CGFloat

File: Fitness/Views/ClassDetail/ClassDetailDescriptionCell.swift (hunk 4 above)
Comment (kevinchan159) [reply Fitness/Views/ClassDetail/ClassDetailDescriptionCell.swift]: not my code but ill fix

File: Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift (hunk 5 above)
Comment (kevinchan159) [reply Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift]: same

File: Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift (hunk 6 above)
Comment (kevinchan159) [reply Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift]: same

File: Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift (hunk 7 above)
Comment (kevinchan159) [reply Fitness/Views/ClassDetail/ClassDetailFunctionCell.swift]: alright u got me

File: Fitness/Views/ClassDetail/ClassDetailHeaderView.swift (hunk 8 above)
Comment (kevinchan159) [reply Fitness/Views/ClassDetail/ClassDetailHeaderView.swift]: yeah we should

File: Podfile (hunk 9 above)
Comment (kevinchan159) [reply Podfile]: its fine to have in master b/c it'll be ignored when you run/archive on Release. i find it useful b/c i don't know what anything is inside the app

**Pull request merged**